python/lsst/cp/pipe/cpDefects.py

```diff
@@ -366,6 +389,97 @@ def _findHotAndColdPixels(self, exp):
         hotPixelCount = {}
         coldPixelCount = {}
 
+        detector = exp.getDetector()
+
+        if self.config.fitAmpGradient:
+            # 1. bin flat
+            # instantiate empty ptc just to use bin_flat
+            ptcForGradientFit = PhotonTransferCurveDataset()
+            # xd and yd in the output table are in pixels in the detector
+            binnedExp = bin_flat(ptcForGradientFit,
+                                 exp,
+                                 bin_factor=self.config.ampGradientBinFactor,
+                                 amp_boundary=self.config.ampGradientBoundarySize,
+                                 apply_gains=False
+                                 )
+
+            # 2. call the fitter
+            transform = detector.getTransform(
+                cameraGeom.PIXELS,
+                cameraGeom.FOCAL_PLANE,
+            )
+
+            xx = np.arange(exp.image.array.shape[1], dtype=np.int64)
+            yy = np.arange(exp.image.array.shape[0], dtype=np.int64)
+            x, y = np.meshgrid(xx, yy)
+            x = x.ravel()
+            y = y.ravel()
+
+            xy = np.vstack((x, y))
+            xf, yf = np.vsplit(transform.getMapping().applyForward(xy.astype(np.float64)), 2)
+            xf = xf.ravel()
+            yf = yf.ravel()
+
+            # we apply the fitting amp by amp
```

Comment:
We measure the gradient on the binned amplifier and apply on the full amplifier, amp by amp.